src/Akeneo/Category/front/src/feature/components/templates/AttributeSettings.tsx

```diff
@@ -15,13 +17,23 @@ export const AttributeSettings = ({attribute, activatedCatalogLocales}: Props) =
   const translate = useTranslate();
   const attributeLabel = getLabelFromAttribute(attribute, userContext.get('catalogLocale'));
   const catalogLocales = useCatalogLocales();
+  const featureFlag = useFeatureFlags();
 
   const [
     isDeactivateTemplateAttributeModalOpen,
     openDeactivateTemplateAttributeModal,
     closeDeactivateTemplateAttributeModal,
   ] = useBooleanState(false);
 
+  const [isRichTextArea, setIsRichTextArea] = useState<boolean>(attribute.type === 'richtext');
+
+  const editAttributeRichTextAreaStatus = useEditAttributeTextAreaStatus(attribute.template_uuid, attribute.uuid);
+
+  const handleRichTextAreaChange = () => {
+    setIsRichTextArea(isRichTextArea => !isRichTextArea);
```

Comment:
```suggestion
    setIsRichTextArea(!isRichTextArea);
```
I would remove the use of the setter callback just to be extra sure that the value is the same between the state update & the api call